src/info/utils/git.rs

```diff
@@ -78,8 +83,38 @@ impl Commits {
             }
         }
 
-        let mut authors_by_number_of_commits: Vec<(Sig, usize)> =
-            author_to_number_of_commits.into_iter().collect();
+        let mut authors_by_number_of_commits: Vec<(Sig, usize)> = email_to_names
+            .into_iter()
+            .map(|(email, name_to_nbr_of_commits)| {
+                let mut author_nbr_of_commits = 0;
+
+                let most_frequent_name = name_to_nbr_of_commits
+                    .into_iter()
+                    .fold(None, |acc, x| {
+                        author_nbr_of_commits += x.1;
+
+                        match acc {
+                            None => Some(x),
+                            Some(e) => {
+                                if x.1 > e.1 {
+                                    Some(x)
+                                } else {
+                                    Some(e)
+                                }
+                            }
+                        }
+                    })
+                    .unwrap();
```

Comment:
I was thinking something kind of like this. Might be some mistakes below. Writing more than a single line in a GitHub suggestion block is a pain :laughing:

Doesn't have to be exactly like this, but destructuring can really help make things more readable than `.0` and `.1`. And if you're mutating something inside of a `.fold`, there's a decent chance that it can be part of the fold.
```suggestion
                let (most_frequent_name, author_nbr_of_commits) = name_to_nbr_of_commits
                    .into_iter()
                    .fold((None, 0), |(common_name, total), (name, commits)| {
                        let common_name = match common_name {
                            Some((n, count) if count > commits => common_name,
                            _ => Some((name, commits)),
                        };
                                
                        (common_name, total + commits)
                    })
                    .unwrap();
```